clippy_utils/src/consts.rs

```diff
@@ -400,6 +400,27 @@ impl<'a, 'tcx> ConstEvalLateContext<'a, 'tcx> {
         let res = self.typeck_results.qpath_res(qpath, id);
         match res {
             Res::Def(DefKind::Const | DefKind::AssocConst, def_id) => {
+                // Check if this constant is based on `cfg!(..)`,
+                // which is NOT constant for our purposes.
+                if let Some(mut node) = self.lcx.tcx.hir().get_if_local(def_id) {
+                    loop {
+                        match node {
+                            Node::Item(&Item {
+                                kind: ItemKind::Const(_, body_id),
```

Comment:
How would code look like that has nested `const` items? I don't see any tests for this. I have a feeling that walking down this chain isn't necessary.